### PR TITLE
build(deps): Bump the Go version to 1.22.7

### PR DESCRIPTION
**Pull request by cason**
Should temporarily solve #4039.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,9 @@
 module github.com/cometbft/cometbft
 
-go 1.22
+go 1.22.5
```
Comment (melekes) [inline go.mod:3]:
```suggestion
go 1.22.7
```

**Review by melekes — APPROVED** [review]

**Comment by cason** [comment]
We need a new version of `cometbft-db` to this to work, but in this case we are using v1.23 indirectly....

**Comment by andynog** [comment]
If we cut a cometbft-db `v0.14.1` bumping the version I think it will unblock this. I've pushed a PR on `cometbft-db` for that

**Comment by melekes** [comment]
@mergify backport v1.x

**Pull request merged**